### PR TITLE
 Add source code for Chapter 11

### PR DESCRIPTION
**Pull request by Peter-VanWeert**

### CODE REVIEW

**Comment by MarcGregoire** [comment]
Mmm, how to verify these source files?
How did you test these?

**Review by MarcGregoire — COMMENTED** [review]
> @MarcGregoire, about half of these examples result in an internal error of the VC++ compiler. True. But the other half do work. As a rule, those where function definitions are in source files can be made to compile with VC++. The others you'll have to verify on sight: you've read the chapter in the book, so you should know all you need to know ;-)

Alright, i'll start with them right away.

File: Exercises/Modules/Chapter 11/Soln11_05/Soln11_05.cpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Splitting a module interface into partitions
```
Comment (MarcGregoire) [inline Exercises/Modules/Chapter 11/Soln11_05/Soln11_05.cpp:1]: I can again not compile this in VC++, but code looks ok.

File: Exercises/Modules/Chapter 11/Soln11_06/words.cppm [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  void show_words(const Words& words);
+}
+
+export namespace w = words;
+
+/* This does not work:
+
+  namespace wrds
+  {
+    // A using directive us not a declaration,
```
Comment (MarcGregoire) [inline Exercises/Modules/Chapter 11/Soln11_06/words.cppm:22]: us=is

File: Exercises/Modules/Chapter 11/Soln11_06/words.cppm [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+export namespace w = words;
+
+/* This does not work:
+
+  namespace wrds
+  {
+    // A using directive us not a declaration, 
+    // and can therefore not be exported from a module.
+    // Namespace aliases are therefore the preferred over
```
Comment (MarcGregoire) [inline Exercises/Modules/Chapter 11/Soln11_06/words.cppm:24]: remove "the"

**Review by MarcGregoire — CHANGES_REQUESTED** [review]

**Comment by MarcGregoire** [comment]
I finished the review of the files.
_**Note:** as already mentioned, I was not able to compile everything. I mentioned this on a few places, but maybe not all places. I suggest you come back to these source files when compilers are up-to-date and verify if everything works correctly at that time!_

File: Examples/Modules/Chapter 11/Ex11_04/Ex11_04.cpp [hunk 4]
```diff
@@ -0,0 +1,10 @@
+// Module implementation partitions
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/Modules/Chapter 11/Ex11_04/Ex11_04.cpp:1]: You need to select Compile as `C++ Module Internal Partition (/internalPartition)` for `roman-internals.cpp`.

File: Examples/Modules/Chapter 11/Ex11_06A/Ex11_06A.cpp [hunk 5]
```diff
@@ -0,0 +1,10 @@
+// Defining and using a namespace
+import <iostream>;
+import squaring;
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/Modules/Chapter 11/Ex11_06A/Ex11_06A.cpp:3]: Missing name of the missing import (<numbers>...)

File: Examples/Modules/Chapter 11/Ex11_08/Ex11_08.cpp [hunk 6]
```diff
@@ -0,0 +1,21 @@
+// Using using declarations and using directives
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/Modules/Chapter 11/Ex11_08/Ex11_08.cpp:1]: This example was not named in the text. Ex11_06B is something else entirely, though.

File: Examples/Modules/Chapter 11/Ex11_08/squaring.cppm [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+export namespace math         // Exports all nested declarations at once
+{
+  
+  using std::numbers::sqrt2;
+  using std::sqrt;    // Never 'using std::sqrt();' or 'using std::sqrt(double);'!
+  auto square(const auto& x) { return x * x; }
+  auto pow4(const auto& x) { return square(square(x)); }
+}
+
+/* Export all names from a namespace from the math namespace using a using directive */
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Examples/Modules/Chapter 11/Ex11_08/squaring.cppm:16]: No, I am exporting all names from the std::numbers namespace...

File: Exercises/Modules/Chapter 11/Soln11_03/Soln11_03.cpp [hunk 8]
```diff
@@ -0,0 +1,26 @@
+// Splitting a module in multiple submodules
```
[the remark this answers is not in the document]
Comment (Peter-VanWeert) [reply Exercises/Modules/Chapter 11/Soln11_03/Soln11_03.cpp:1]: Compiles just fine if you replace all imports with #includes

File: Exercises/Modules/Chapter 11/Soln11_05/Soln11_05.cpp (hunk 1 above)
Comment (Peter-VanWeert) [reply Exercises/Modules/Chapter 11/Soln11_05/Soln11_05.cpp:1]: Even after fixing the circular include, I could still not compile it...

**Pull request merged**